### ISSUE TITLE
Not resolving `tsconfig` paths

### ISSUE THREAD

**Issue by SimonCockx** [import-js/eslint-plugin-import#2652]
I have two projects, `project-a` and `project-b`, one importing the other via a `paths` synonym defined in `tsconfig.base.json`. Although the TypeScript compiler knows how to resolve the import, this plugin is complaining that it cannot resolve the path.

I've read the documentation of both this plugin and the TypeScript resolver plugin, and I think I configured everything right, but with no luck. Is there something wrong with my setup?

Minimal repro:
- clone https://github.com/SimonCockx/eslint-import-bug
- `cd ./eslint-import-bug`
- `npm install`
- `npx eslint project-a/src/index.ts`

Snippets:

`tsconfig.base.ts`
```
"compilerOptions": {
  "paths": {
    "@importbug/b": ["project-b/src/index.ts"]
  },
  ...
}
```

`project-a/.eslintrc.json`
```
{
  "plugins": [
    "@typescript-eslint",
    "import"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": ["project-a/tsconfig.json"],
    "sourceType": "module"
  },
  "settings": {
    "import/resolver": {
      "typescript": {
        "alwaysTryTypes": true,
        "project": "tsconfig.json",
        "extensions": [".ts"]
      }
    },
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts", ".tsx"
      ]
    }
  },
  "extends": [
    "plugin:import/typescript", "plugin:import/recommended"
  ]
}
```

`project-a/src/index.ts`
```
import { b } from "@importbug/b";
                   ^^^^^^^^^^^
b;
```

**Comment by ysulyma** [comment]
Also encountering this

**Comment by ljharb** [comment]
Try extending `plugin:import/typescript`?

**Comment by SimonCockx** [comment]
@ljharb That's already the case - see this line: https://github.com/SimonCockx/eslint-import-bug/blob/main/project-a/.eslintrc.json#L26.

**Comment by SimonCockx** [comment]
Interestingly, I just found out that running `eslint` succeeds if I `cd` into `project-a` and change
```
"parserOptions": {
    "project": ["project-a/tsconfig.json"],
    "sourceType": "module"
  },
```
into
```
"parserOptions": {
    "project": ["tsconfig.json"],
    "sourceType": "module"
  },
```

Could this be a bug with the way that `eslint` handles configuration that is not in the project root?

**Comment by ljharb** [comment]
We're using tsconfig-paths v3, which doesn't support "extends", i believe - that might be relevant.

**Comment by SimonCockx** [comment]
@ljharb Not sure how that's relevant for the issue I've described. I don't use `tsconfig-paths`, and I think eslint acts directly on the source files, so I assume it doesn't care about the problem that `tsconfig-paths` sets out to solve.

**Comment by ljharb** [comment]
@SimonCockx eslint-plugin-import uses tsconfig-paths in order to resolve typescript settings.

**Comment by zeorin** [comment]
If you're using Yarn, you can use [resolutions](https://yarnpkg.com/cli/set/resolution) to upgrade `tsconfig-paths`: `yarn set resolution tsconfig-paths@npm:^3.14.2 ^4`.

**Comment by JounQin** [comment]
1. Use `https://github.com/un-es/eslint-plugin-i`
2. Use `yarn set resolution tsconfig-paths ^4` like above as @zeorin suggested, or https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides

**Comment by ljharb** [comment]
That’s not an official solution.

**Comment by ljharb** [comment]
That remains open, and if an issue (such as this one) has no solution it should remain open indefinitely.
